leveldb-server/src/db.ts

```diff
@@ -125,9 +125,7 @@ export class Handler implements DBHandler {
       return JSON.parse(val.toString());
     } catch (err) {
       if (err.notFound) return undefined;
-      err.debugId = debugId;
-      err.message = `[${debugId}] ${err.message}`;
-      throw err;
+      throw { ...err, debugId, message: `[${debugId}] ${err.message}` };
```

Comment:
Although this is legal, I suggest we avoid `throw`ing non-`instanceof Error`s and have a tiny `class DBError extends Error` for all `toString()`, `stack` goodiness.

Reply:
Here's the thing.  If I add an `Error`, its constructor generates a new stacktrace -- which is all the boring bits.  And this isn't Java, we don't have proper exception chaining.
So I'm doing it using _fake_ exception chaining.  It's not like JavaScript `Error`s have any useful fields other than `message`, `type` and `stacktrace`.